### ISSUE TITLE
Print document

### ISSUE THREAD

**Issue by martinlindhe** [atom/atom#3525]
Would be nice to have a Print feature. Unless I'm mistaken, there is no such menu option or plugin.

**Comment by NatalieWolfe** [comment]
:+1:

**Comment by lee-dohm** [comment]
@postcasio created a [package for printing from Atom](https://discuss.atom.io/t/printing-support/760/24?u=leedohm).

**Comment by James-Parsons** [comment]
Is any work being done on this?

**Comment by mnquintana** [comment]
As far as I can tell, this is blocked on atom/electron#865, and likely needs atom/electron#805 and #1157 to be sufficient. There's a nice summary at: https://github.com/github/gsoc#improve-printing-support-of-atom-shell

There is also a thread on Atom Discuss for more background: https://discuss.atom.io/t/printing-support/760

**Comment by bryanhanson** [comment]
And there had been a 'working very well not quite perfect in all cases' package here: https://atom.io/packages/atom-printer <https://atom.io/packages/atom-printer> but it is gone for now.  I have it on my computer and so it still works.  Looks like a more official printing stuff is coming soon.

**Comment by di48l069** [comment]
would also be nice if you guys can also implement github markup in the print feature where for example you can use ##header and it make it a header i dunno just example

**Comment by pozgo** [comment]
Any update on this feature? 
Thanks

**Comment by jorymil** [comment]
@bryanhanson Would you mind publishing the atom-printer plugin?  I'd certainly like a copy, even if it's no longer official.

**Comment by bryanhanson** [comment]
Unfortunately, it no longer works for me, apparently some changes as Atom went to 1.0 broke it. Plus, it’s not my code and my recent attempts to prompt the original author for a status report of any kind went unreturned.

**Comment by bryanhanson** [comment]
I found the code, and the license permits it, so it is now here: https://github.com/bryanhanson/atom-printer

Anyone who can fix it up will be thanked!

**Comment by di48l069** [comment]
Any update on this feature? @lee-dohm

**Comment by lee-dohm** [comment]
Other than what is mentioned in this Issue, no.

**Comment by pgnd** [comment]
Fwiw, a similar "discussion"
  http://sublimetext.userecho.com/topic/25170-printing/
re: SublimeText's (then) unique lack of printing support, and lots of well-articulated arguments.

**Comment by jeffmcneill** [comment]
I don't want to print so much as export to PDF, but it is kind of the same thing. I agree with the shouldn't have to print part, but there are people I interact with in the universe who demand printable, if not printed, material, on occasion.

**Comment by ufechner7** [comment]
As far as I can see the printing bugs in electron are fixed. Furthermore there is an updated printing plugin here: https://github.com/palant/atom-printer , but it is not registered and I don't know how to test it. Any hints?

**Comment by bryanhanson** [comment]
@ufechner7 Thanks for noticing this.  I had previously tested this code and it did not work.  However, I just installed it and it seems to work now, I guess due to changes in electron made in the meantime.  I'm on Mac, I just copied the files into `/Users/bryanhanson/.atom/packages/atom-printer` and it works now!  So you just need to find the corresponding directory on your system and hopefully it will work for you.  Thanks again for the tip.

**Comment by jpluimers** [comment]
@ufechner7 By name print-atom looks promising, but contrary to the advertised "Print current atom window using print dialog", it prints the whole Atom UI (Mac OS X; tested in atom 1.6.2 and 1.7.2 from https://atom.io/download/mac), which is of no use at all for me.

export-html works better: then I can print from my web-browser. Not perfect, but workable.

**Comment by RickBross** [comment]
+1

**Comment by LeadManPL** [comment]
3 years passing soon... any chance this "feature" will be implemented?

**Comment by bryanhanson** [comment]
There is no development going on here AFAIK.  I haven't searched for alternatives lately, maybe someone else is working on it.   If you follow all the various threads, I'm pretty sure there is no intent to make printing a standard feature of Atom.  The code does still mostly work.  I just tested it, it prints fine but now only in b/w, no color.  It used to be in color.  Not bad for stale code.

**Comment by lee-dohm** [comment]
Thanks for the ping.

This isn't something that the Atom maintainer team is going to implement. If people want to build packages that offer this feature and need additional API support in order to make that happen, we would consider it. However, printing support itself isn't a feature that we're going to maintain in core.

Thanks again for your interest!

**Comment by LeadManPL** [comment]
@bryanhanson : Could you provide name of this alternative that is able to print BW?

**Comment by bryanhanson** [comment]
It's referenced [above](https://github.com/palant/atom-printer).  I did have a gist of it (also mentioned above) but I took it down.  My comment above on March 13, 2016 tells where to put it, at least on Mac, should be an analogous path on other platforms.

**Comment by jeffmcneill** [comment]
There are several plugins that can print, such as:
- https://atom.io/packages/print-code (contents of the text editor, with line numbers)
- https://atom.io/packages/print-atom (full screen, including project pane, all tab heads, and current tab - buggy implementation)

**Comment by jpluimers** [comment]
Reminder to self to try https://github.com/postcasio/atom-printer

**Comment by bryanhanson** [comment]
Thank you @jeffmcneill for the tip on `print-code` it works well!
